src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService.cs

```diff
@@ -28,6 +28,24 @@ public CSharpIntroduceVariableService()
         {
         }
 
+        protected override bool IsConstantDefinition(ExpressionSyntax expressionSyntax)
+        {
+            if (expressionSyntax.WalkUpParentheses() is not
+                {
+                    Parent: EqualsValueClauseSyntax
+                    {
+                        Parent: VariableDeclaratorSyntax declarator
+                    }
+                })
+            {
+                // Sub-expression
+                return false;
+            }
+
+            return declarator.GetAncestorOrThis<LocalDeclarationStatementSyntax>()?.IsConst == true
+                || declarator.GetAncestorOrThis<FieldDeclarationSyntax>()?.Modifiers.Any(SyntaxKind.ConstKeyword) == true;
```

Comment:
so i think all this code can be moved into the base class.  from there you can use ISyntaxFacts and/or SyntaxGenerator.  Effectively, you just want to get get the parent of the expr, see if you're in a var-decl, and if so, grab hte modifiers from it (i.e. SyntaxGenerator.GetModifiers).   Then you can just check if the decl is const and  bail.

Reply:
I can't find a handful way to check if current node is the body of variable initializer, and get the corresponding variable. Even the IOperation tree of two languages are not identical.

And also, this refactoring providing hasn't been using ISyntaxFacts and related.

Reply:
@CyrusNajmabadi ping for my PRs assigned to you